mango-viz/bdgenomics/mango/io/VCFfile.py

```diff
@@ -0,0 +1,113 @@
+#
+# Licensed to Big Data Genomics (BDG) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional insourceion
+# regarding copyright ownership.  The BDG licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from .genomicfile import GenomicFile
+from collections import OrderedDict
+import gzip
+
+class VCFFile(GenomicFile):
+
+    global VCF_HEADER
+    global NUM_SAMPLES
+    VCF_HEADER = ['CHROM', 'POS', 'ID', 'REF', 'ALT', 'QUAL', 'FILTER', 'INFO', 'FORMAT']
+
+
+    @classmethod
+    def _read(cls, filename, large = True):
+        """Open an optionally gzipped VCF file and return a pandas.DataFrame with
+        each INFO field included as a column in the dataframe.
+        
+        Borrowed from https://gist.github.com/slowkow/6215557
+
+        Note: Using large=False with large VCF files. It will be painfully slow.
+        :param filename:    An optionally gzipped VCF file.
+        :param large:       Use this with large VCF files to skip the ## lines and
+                            leave the INFO fields unseparated as a single column.
+        """
+
+        if large:
+            # Set the proper argument if the file is compressed.
+            comp = 'gzip' if filename.endswith('.gz') else None
+            # Count how many comment lines should be skipped.
+            NUM_COMMENTS = cls._count_comments(filename)
+            # Return a simple DataFrame without splitting the INFO column.
+            df = cls.dataframe_lib.read_table(filename, compression=comp, skiprows=range(NUM_COMMENTS-1))
+            NUM_SAMPLES = len(df.columns) - len(VCF_HEADER)
+            df.columns = VCF_HEADER + [i for i in range(1, NUM_SAMPLES+1)]
+            return df       
+
+    @classmethod
+    def _getcallinformation(cls, row):
+        sample_column_names = row.index[9:]
+        call_information = '"calls":['
```

Comment:
This doesn't look like the bdg-formats `Variant` or `Genotype` schema, or even similar to ADAM `VariantContext` or htsjdk `VariantContext`.  I assume the schema here is most closely aligned with the GA4GH schema, for sending to pileup?

Reply:
Yes, the schema is aligned with the ga4gh schema.

Reply:
👍